src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -161,6 +164,12 @@ public class KubernetesCloud extends Cloud implements PodTemplateGroup {
     @CheckForNull
     private GarbageCollection garbageCollection;
 
+    /**
+     * namespace -> informer
+     * Use to watch pod events per namespace.
+     */
+    private transient Map<String, SharedIndexInformer<Pod>> informers = new ConcurrentHashMap<>();
```

Comment:
Need to be careful on what would happen to these informers if/when a `KubernetesCloud` gets removed.

Reply:
AFAICT the `informers` map would be unreachable and it would be garbage collected, so the `DefaultSharedIndexInformer` would be collected too and with it the `informerExecutor` (each instance has an executor), so the resync would be stop happening (I've checked it actually stops reporting events).

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/1627/commits/66e538efbc67d76c450f5ae9b1b9c5e590378558 -> for thread safety.